src/Service.Tests/Unittests/RuntimeConfigLoaderJsonDeserializerTests.cs

```diff
@@ -128,6 +128,51 @@ public void CheckCommentParsingInConfigFile()
             Assert.IsTrue(RuntimeConfigLoader.TryParseConfig(actualJson, out RuntimeConfig _), "Should not fail to parse with comments");
         }
 
+        /// <summary>
+        /// Test to validate that optional properties
+        /// are nullable, don't contain defaults on serialization
+        /// but have the effect of default values when deserialized.
+        /// It starts with a minimal config and incrementally
+        /// adds the optional subproperties. At each step, tests for valid deserialization.
+        [TestMethod]
+        public void TestNullableOptionalProps()
+        {
+            // Test with no runtime property
+            StringBuilder minJson = new(@"
+                                ""data-source"": {
+                                    ""database-type"": ""mssql"",
+                                    ""connection-string"": ""@env('test-connection-string')""
+                                    },
+                                ""entities"": { }");
+
+            TryParseAndAssertOnDefaults("{" + minJson + "}", out _);
+
+            // Test with an empty runtime property
+            minJson.Append(@", ""runtime"": ");
+            string emptyRuntime = minJson + "{ }}";
+            TryParseAndAssertOnDefaults("{" + emptyRuntime, out _);
+
+            // Test with empty sub properties of runtime
+            minJson.Append(@"{ ""rest"": { }, ""graphql"": { },
+                            ""base-route"" : """",");
+            StringBuilder minJsonWithHostSubProps = new(minJson + @"""telemetry"" : { }, ""host"" : ");
+            StringBuilder minJsonWithTelemetrySubProps = new(minJson + @"""host"" : { }, ""telemetry"" : ");
+
+            string emptyRuntimeSubProps = minJsonWithHostSubProps + "{ } } }";
+            TryParseAndAssertOnDefaults("{" + emptyRuntimeSubProps, out _);
+
+            // Test with empty host sub-properties
+            minJsonWithHostSubProps.Append(@"{ ""cors"": { }, ""authentication"": { } } }");
+            string emptyHostSubProps = minJsonWithHostSubProps + "}";
+            TryParseAndAssertOnDefaults("{" + emptyHostSubProps, out _);
+
+            // Test with empty telemetry sub-properties
+            minJsonWithTelemetrySubProps.Append(@"{ ""application-insights"": { } } }");
+
+            string emptyTelemetrySubProps = minJsonWithTelemetrySubProps + "}";
+            TryParseAndAssertOnDefaults("{" + emptyTelemetrySubProps, out _);
+        }
```

Comment:
I'm not following what the test is doing with empty object's `{ }` for the properties `telemetry` and `application-insights`. There is no reference in the `TryParseAndAssertOnDefaults(..)` to those properties.

Are you just ensuring that the properties set to empty object `{}` result in successful config parsing due to default values and nullable optional properties... as opposed to null ref exceptions?

Reply:
yes, Im ensuring there are no null ref exceptions. Also added a test for checking disabled `AppInsights`